ssh/basic-sshd.go

```diff
@@ -228,15 +230,21 @@ func (s *sshRelay) handleChannel(ctx context.Context, wg *sync.WaitGroup, newCha
 	// channel type of "session". The also describes
 	// "x11", "direct-tcpip" and "forwarded-tcpip"
 	// channel types.
-	if t := newChannel.ChannelType(); t != "session" {
+	switch newChannel.ChannelType() {
+	case "session":
+		s.handleChannelTypeSession(ctx, wg, newChannel, namespace, userID)
+	case "direct-tcpip":
+		s.handleChannelTypeDirectTCPIP(ctx, wg, newChannel, namespace, userID)
+	default:
 		s.log.Error("unknown channel type", zap.String("type", newChannel.ChannelType()))
-		err := newChannel.Reject(ssh.UnknownChannelType, fmt.Sprintf("unknown channel type: %s", t))
+		err := newChannel.Reject(ssh.UnknownChannelType, fmt.Sprintf("unknown channel type: %s", newChannel.ChannelType()))
 		if err != nil {
 			s.log.Error("failed to reject channel", zap.Error(err))
 		}
-		return
 	}
+}
 
+func (s *sshRelay) handleChannelTypeSession(ctx context.Context, wg *sync.WaitGroup, newChannel ssh.NewChannel, namespace, userID string) {
```

Comment:
I think this function is broken.

Reply:
Handle session should work. At least I can ssh through the ssh agent into a kubernetes pod. 
Direct-tcpip is not working since there are some issues regarding the port forwarding.

Reply:
Ah, I was just guessing because of the linting warning. 
Go on then but fix the lint errors :P

Reply:
Now it works, but the code quality sucks and the sever needs some refactoring for properly handing the connection requests.